### PR TITLE
Fix autosave/dialog flow

### LINKED ISSUE #12474: Auto-saving: saving after auto-saving causes an error
## Bug Description
Looks like after auto-save happens on a published story, saving causes an error:

Screenshot 2022-10-14 at 12 17 24

## Steps to Reproduce
1. Save a story
2. Make some changes
3. Wait until auto-save happens
4. Now click "Update"
5. See error happening and the editor crashing

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by merapi**
## Summary

This PR will fix autosave triggered on every element update and 2 issues from [here](https://xwp.slack.com/archives/C013J2EGH8R/p1666172443879059).

There is one more [issue](https://github.com/GoogleForCreators/web-stories-wp/issues/12474#issuecomment-1281083460) with autosave but I need to think about it a little more, it can have a bigger impact on the app (`hasNewChanges` is used in a few places).

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12474

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]
At first glance looks good 👍 

Dig the improved naming

**Comment by kkalarickal** [comment]
@merapi @swissspidy -- I can confirm that the 2nd auto-save triggers after making further changes to the webstory in draft. It happens within 60 sec of the last edit.

**Comment by kkalarickal** [comment]
These are the sanity workflows that I can confirm work:

1. single session, saving local session auto-draft and restoring without saving-draft 
2. single session, saving local sesion auto-post and restoring a draft without saving to db
3. multiple sessions, saving to db, and opening in another browser/login
4. multiple sessions, published story, and restoring from local changes
 - there was a problem restoring the latest changes #12547
5. concurrent sessions, take over of draft story with changes
6. concurrent session, take over of published story with changes

File: includes/templates/admin/edit-story.php [hunk 1]
```diff
@@ -88,6 +88,11 @@
 			'type'    => $post_type_object->name,
 		]
 	),
+	"/web-stories/v1/$stories_rest_base/{$post->ID}/lock/?" . build_query(
+		[
+			'_embed' => 'author',
+		]
+	),
```
Comment (spacedmonkey) [inline includes/templates/admin/edit-story.php]: This was not preloaded by design. Why is it being preloaded now?

Comment (swissspidy) [reply includes/templates/admin/edit-story.php]: See discussion at https://xwp.slack.com/archives/C01H2K83WD9/p1666788696201389 for an explanation.

The aim is to immediately get the current lock owner information right when opening the editor, without having to wait for the 60s interval to pass first. There's no harm in preloading the response and saving 1 HTTP request.

Note that preloading a response only stores it once. Any subsequent calls will trigger an HTTP request as expected. Just the first one is fetched from cache.

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
Seems like now, I am not seeing the message that says, the story has been taken over. Now, I see, when a story is taken over, I always see this message. 
Screenshot 2022-10-26 at 16 22 23

So this is how it should work. 

1. User 1, edits story. Sets lock. 
2. User 2, open story. 
3. User 2, takes over story. 
4. User 1, see message that story was taken over, without option to over story back over.

**Comment by spacedmonkey** [comment]
https://user-images.githubusercontent.com/237508/198068785-e2b231b9-3cfb-40cf-b6ea-a6c584433b16.mov

See is now a tag of war between the two users. This is not how it used to be.

**Comment by merapi** [comment]
> Seems like now, I am not seeing the message that says, the story has been taken over. Now, I see, when a story is taken over, I always see this message. Screenshot 2022-10-26 at 16 22 23
> 
> So this is how it should work.
> 
> 1. User 1, edits story. Sets lock.
> 2. User 2, open story.
> 3. User 2, takes over story.
> 4. User 1, see message that story was taken over, without option to over story back over.

Reverted the breaking change.

**Pull request merged**